queries/apex/highlights.scm

```diff
@@ -0,0 +1,340 @@
+(type_arguments (type_identifier) @type)
+
+(field_access
+  field: (identifier) @property)
+
+((scoped_identifier
+  scope: (identifier) @type)
+ (#match? @type "^[A-Z]"))
+((method_invocation
+  object: (identifier) @type)
+ (#match? @type "^[A-Z]"))
+
+
+(field_declaration
+  type: (type_identifier) @type)
+
+(method_declaration
+  (formal_parameters
+    (formal_parameter
+      name: (identifier) @parameter)))
+
+(formal_parameter
+  type: (type_identifier) @type
+  (identifier) @variable)
+
+(enhanced_for_statement
+  type: (type_identifier) @type
+  name: (identifier) @variable )
+
+(enhanced_for_statement
+  value: (identifier) @variable)
+
+(enhanced_for_statement
+  name: (identifier) @variable)
+
+(object_creation_expression
+  type: (type_identifier) @type)
+
+(array_creation_expression
+  type: (type_identifier) @type)
+
+(array_type
+  element: (type_identifier) @type)
+
+(catch_formal_parameter
+  (type_identifier) @type
+  name: (identifier) @variable)
+
+(return_statement
+  (identifier) @variable)
+
+(local_variable_declaration
+  (variable_declarator
+    name: (identifier) @variable ))
+
+(for_statement
+  condition: (binary_expression
+    (identifier) @variable))
+
+(for_statement
+  update: (update_expression
+    (identifier) @variable))
+
+(constructor_declaration
+  name: (identifier) @constructor)
+
+(dml_type) @function.builtin;;
+
+(bound_apex_expression
+  (identifier) @variable)
+
+(assignment_operator) @operator
+
+(update_expression ["++" "--"] @operator)
+
+(instanceof_expression
+  left: (identifier) @variable
+  right: (type_identifier) @type )
+
+(cast_expression
+  type: (type_identifier) @type
+  value: (identifier) @variable)
+
+(switch_expression
+  condition: (identifier) @variable)
+
+(switch_label
+  (type_identifier) @type
+  (identifier) @variable )
+
+(switch_rule
+  (switch_label
+    (identifier) @variable ))
+
+(trigger_declaration
+  name: (identifier) @type
+  object: (identifier) @type
+  (trigger_event) @keyword
+  ("," (trigger_event) @keyword)*)
+
+"@" @operator
+
+(binary_expression
+  operator: [
+    ">"
+    "<"
+    ">="
+    "<="
+    "=="
+    "==="
+    "!="
+    "!=="
+    "&&"
+    "||"
+    "+"
+    "-"
+    "*"
+    "/"
+    "&"
+    "|"
+    "^"
+    "%"
+    "<<"
+    ">>"
+    ">>>"] @operator)
+
+(binary_expression
+  (identifier) @variable)
+
+(unary_expression
+  operator: [
+    "+"
+    "-"
+    "!"
+    "~"
+  ]) @operator
+
+(map_initializer "=>" @operator)
+
+[
+  (boolean_type)
+  (void_type)
+] @type.builtin;;
+
+; Fields
+
+(field_declaration
+  declarator: (variable_declarator
+    name: (identifier) @field))
+
+(field_access
+  field: (identifier) @field)
+
+; Variables
+
+(field_declaration
+  (modifiers (modifier ["final" "static"])(modifier ["final" "static"]))
+  (variable_declarator
+    name: (identifier) @constant))
+
+(variable_declarator
+  (identifier) @property)
+
+((identifier) @constant
+  (#match? @constant "^_*[A-Z][A-Z\\d_]+$"))
```

Comment:
We prefer lua-match wherever possible

Check section 5.4.1 of luadoc, named lua-patterns

Reply:
solved with latest commit